### PR TITLE
A32: Implement VCVTT, VCVTB

### LINKED ISSUE #3705: Unimplemented VCVTB.F16 instruction
Requires the VCVTB.F16 instruction.</br>

```
The instruction at 0x0000000000242824 (opcode 0xeeb31a41) is undefined!
```

Description:
```
vcvtb.f16.f32 s2, s2
```
### Required by: 
Radiant Silvergun

### PR DESCRIPTION
**Pull request by merryhime**
Should close #3705.

### CODE REVIEW

File: ARMeilleure/Decoders/OpCode32SimdCvtTB.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        public new static OpCode Create(InstDescriptor inst, ulong address, int opCode) => new OpCode32SimdCvtTB(inst, address, opCode, false);
+        public static OpCode CreateT32(InstDescriptor inst, ulong address, int opCode) => new OpCode32SimdCvtTB(inst, address, opCode, true);
+
+        public OpCode32SimdCvtTB(InstDescriptor inst, ulong address, int opCode, bool isThumb) : base(inst, address, opCode)
+        {
+            IsThumb = isThumb;
+
+            Op   = ((opCode >> 16) & 0x1) != 0;
+            T    = ((opCode >> 7) & 0x1) != 0;
+            Size = ((opCode >> 8) & 0x1) != 0 ? 3 : 2;
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCode32SimdCvtTB.cs:20]: This could probably be a 1 bit value to make it a bit simpler (Size = 1 being double, and 0 single).

**Review by gdkchan — APPROVED** [review]
LGTM, thanks. Would be nice to have a fast path using F16C like we do for Arm64.

**Comment by merryhime** [comment]
Pushed f16c implementation; probably can be slightly improved but my brain is non-functional atm.

**Comment by EmulationFanatic** [comment]
Radiant Silvergun seems to be working fine with the PR; it would crash shortly after starting the first level on Master. Here is the game proceeding on to the second level.
image

**Review by AcK77 — APPROVED** [review]

**Comment by TRoehrman** [comment]
I just wanted to say, you have solved a decades long problem, that of getting Radiant Silvergun HD to work on PC. I've been waiting since Ikaruga HD. My absolute sincerest gratitude goes to you, MerryHime, and everyone who finally made this possible after more than a decade of waiting.

Thank you, from the bottom of my gaming heart.

Be attitude for gains!

**Pull request merged**